lib/drivers/MongoDriver.ts

```diff
@@ -104,14 +104,35 @@ export class MongoDriver extends DatabaseDriver<MongoConnection> {
     await Promise.all(promises);
   }
 
-  private createIndexes(meta: EntityMetadata, type: 'indexes' | 'uniques') {
+  private createIndexes(meta: EntityMetadata) {
     const promises: Promise<string>[] = [];
+    meta.indexes.forEach(index => {
+      let fieldOrSpec: string | any;
+      const properties = Utils.flatten(Utils.asArray(index.properties).map(prop => meta.properties[prop].fieldNames));
+      if (index.type === 'text') {
+        const spec: {[key: string]: 'text'} = {};
+        properties.forEach(prop => spec[prop] = 'text');
+        fieldOrSpec = spec;
+      } else {
+        fieldOrSpec = properties;
+      }
+      promises.push(this.getConnection('write').getCollection(meta.name).createIndex(fieldOrSpec, {
+        name: index.name,
+        unique: index.type === 'uniques',
```

Comment:
this is wrong, `index.type` is not the same as the parameter `type` you just removed


```suggestion
        unique: false,
```